packages/common/conductor/middleware/s3TaskDataFetcher.ts

```diff
@@ -35,7 +35,10 @@ const formatErrorMessages = (schema: string, error: z.ZodError): string[] =>
     return `${schema} parse error. Schema mismatch`
   })
 
-const s3TaskDataFetcher = <T>(schema: z.ZodSchema, handler: Handler<TaskDataInputData<T>>): OriginalHandler => {
+const s3TaskDataFetcher = <T>(
+  schema: z.ZodSchema | ((taskData: unknown) => z.ZodSchema),
```

Comment:
`s3TaskDataFetcher` accepts both ZodSchema and a function that returns ZodSchema.